lib/active_merchant/billing/gateways/payeezy.rb

```diff
@@ -119,27 +135,29 @@ def add_authorization_info(params, authorization)
         params[:method] = method
       end
 
+      def add_creditcard_for_tokenization(params, payment_method)
+        params[:apikey] = PARAMS_FOR_TOKENIZATION[:api_key]
+        params[:js_security_key] = PARAMS_FOR_TOKENIZATION[:js_security_key]
+        params[:ta_token] = PARAMS_FOR_TOKENIZATION[:ta_token]
+        params[:callback] = PARAMS_FOR_TOKENIZATION[:callback]
+        params[:type] = PARAMS_FOR_TOKENIZATION[:type]
+        params['credit_card.type'] = CREDIT_CARD_BRAND[payment_method.brand]
+        params['credit_card.cardholder_name'] = payment_method.name
+        params['credit_card.card_number'] = payment_method.number
+        params['credit_card.exp_date'] = format_exp_date(payment_method.month, payment_method.year)
+        params['credit_card.cvv'] = payment_method.verification_value if payment_method.verification_value?
+      end
+
       def add_payment_method(params, payment_method, options)
         if payment_method.is_a? Check
           add_echeck(params, payment_method, options)
+        elsif payment_method.is_a? String
```

Comment:
Hmm. If reference purchases were refer implemented, this could collide with those, since they are not tokens. Maybe premature optimization.

Reply:
I'm okay with worrying about this when reference purchases are requested.

Reply:
👍